server/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -90,7 +91,7 @@
     public static final boolean DEFAULT_FAIL_ON_UNSUPPORTED_FIELDS = true;
 
     private static final Set<Class<? extends MappedFieldType>> SUPPORTED_FIELD_TYPES = new HashSet<>(
-            Arrays.asList(TextFieldType.class, KeywordFieldType.class));
+            Arrays.asList(TextFieldType.class, KeywordFieldType.class, AnnotatedTextFieldType.class));
```

Comment:
Another example of core hard-coding text types

Reply:
I agree supporting more-like-this would be nice ultimately, but let's just skip it for now?

Reply:
I was trying to workaround the lack of MLT using the TermVectorsService but encountered a similar hard-coded road-block: https://github.com/elastic/elasticsearch/blob/00b0e1006320cf8d11d2aaf2f4233dab1bcc1be3/server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java#L165